### PR TITLE
Project 7: Validations, Model specs

### PR DESCRIPTION
**Pull request by er-contreras**
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
 - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all your controllers.

                            ** Run db:setup please! **

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@ class User < ApplicationRecord
   has_many :comments
   has_many :likes
 
+  validates :name, presence: true
+  validates :posts_count, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
```
Comment (sinansevgi) [inline app/models/user.rb:11]: - [x] You are displaying the users' posts with the most recent comments on the `post` index page. Looks like you are making two different queries for this and this is causing an N+1 problem.  You can include comments to your post query to solve this N+1 issue. 

Please check the below screenshots

Screenshot from 2021-12-03 20-55-49
Currently, the application makes 4 SQL queries, 1 for posts and 3 for comments of each post. This significantly reduces your application performance (Total query time 12.3 ms)

Screenshot from 2021-12-03 20-59-45
If you solve the N+1 issue your query will look like this, just two queries for posts and comments. (Total query time 4.2ms it is **3** times faster than the query with N+1 issue)

**Review by sinansevgi — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
------

Hi @eri8-9, 

Good job so far!
Good job with the validations :+1: 
There are some issues that you still need to work on to go further but you are almost there!


 ## Required Changes ♻️
_Please check the inline comments for my suggestions_ :arrow_down:

## Optional suggestions
- [ ] **You should avoid calling model methods from your views. The views should get their data from the controller, They should not interact with the model.**

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (DeliceLydia) [reply app/models/user.rb:11]: Addressed this!

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉

## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**